broker/core/src/main/groovy/com/swisscom/cloud/sb/broker/backup/shield/ShieldApiException.java

```diff
@@ -0,0 +1,36 @@
+package com.swisscom.cloud.sb.broker.backup.shield;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.client.HttpStatusCodeException;
+
+public class ShieldApiException extends RuntimeException {
+    private final String description;
+    private final HttpStatus httpStatus;
+
+    private ShieldApiException(String description, Throwable ex) {
+        super(description, ex);
+        this.description = description;
+        httpStatus = (ex instanceof HttpStatusCodeException) ? ((HttpStatusCodeException) ex).getStatusCode(): null;
+    }
+
+    public static ShieldApiException of(String description, Throwable ex) {
+        return new ShieldApiException(description, ex);
+    }
+
+    public boolean isHttpStatusCodeException() {
+        return httpStatus != null;
+    }
+
+    public boolean isNotFound() {
+        return isHttpStatusCodeException() && (httpStatus.value() == 404) || httpStatus.value() == 501;
+    }
+
+    public String toString() {
```

Comment:
Bad `toString` . IMHO: 
```suggestion
    public String toString() {
               return String.format("ShieldApiException[%s %s: %s]",
                                                  isHttpStatusCodeException() ? httpStatus.value() : "-",
                                                  description != null ? description : "no description",
                                                  getCause() != null ? getCause().getMessage() : "no cause");
```
will be better because is human readable and much easier to be parsed with a script. But what about an immutable value so you don't have to make those checks of nullity? An exception is a perfect candidate for being immutable.